### DISCUSSION TITLE
Can I change the loading of .env files on a per-environment basis?

### DISCUSSION THREAD

**Discussion by kazuyoshi80** [nuxt/nuxt#17071] [Questions]
# Can I change the loading of .env files on a per-environment basis?

Is it possible to prepare .env files according to the execution environment as shown below and change the .env file to be read according to the value of process.env.ENV?

```
config/.env.local
config/.env.dev
config/.env.production
```

In Nuxt2, the following settings were used.

```
const envPath = `config/.env.${process.env.ENV || 'local'}`
require('dotenv').config({ path: envPath })
const nuxtConfig: NuxtConfig = {
  ...
  dotenv: {
    filename: envPath,
  },
}
```

**Comment by kazuyoshi80** [comment]
Can Nuxt3 change the PATH of the .env file as follows?

```
require('dotenv').config({ path: '/custom/path/to/.env' })
```

https://github.com/motdotla/dotenv#path

**Reply by kazuyoshi80** [reply]
In my case, I made sure to change the .env file to match the environment to be built before building

**Comment by TheAlexLichter** [comment]
From my understanding, there is no such way built into Nuxt (yet?). According to a search for `.env`, the only occurence is in the `preview` command which loads only the main `.env` file.

https://github.com/nuxt/framework/blob/e0b2a3bfe4ebc2243998cf82630577a7a4114253/packages/nuxi/src/commands/preview.ts#L38-L41

If I haven't missed anything, this should be a feature request 😋 

PS: Be aware that the `.env` file is only loaded during preview and build, not when running the server in production (as described [in the docs](https://github.com/nuxt/framework/blob/9a0fc577247b6c697908b371f6b7ec6d293d29ec/docs/content/3.api/5.commands/preview.md))

**Reply by zyyv** [reply]
> From my understanding, there is no such way built into Nuxt (yet?). According to a search for `.env`, the only occurence is in the `preview` command which loads only the main `.env` file.
> 
> https://github.com/nuxt/framework/blob/e0b2a3bfe4ebc2243998cf82630577a7a4114253/packages/nuxi/src/commands/preview.ts#L38-L41
> 
> If I haven't missed anything, this should be a feature request 😋
> 
> PS: Be aware that the `.env` file is only loaded during preview and build, not when running the server in production (as described [in the docs](https://github.com/nuxt/framework/blob/9a0fc577247b6c697908b371f6b7ec6d293d29ec/docs/content/3.api/5.commands/preview.md))

I found that in #3751, migrating config to `@nuxt/schema`, but now there is no `dotenv` configure in `@nuxt/schema`. Is that intentional?

**Comment by sxgrant** [comment]
The docs claim that .env variants are supported, although I feel that I've just wasted a bunch of time trying to get that to work.
https://nuxtjs.org/docs/directory-structure/nuxt-config#env-support

Is this just a documentation error?

```
You can customize the path by using --dotenv <file> or disable entirely with --dotenv false. For example, you might specify a different .env file in production, staging or development environments.
```

These features (disabling dotenv, or pointing to an alternate path) don't seem to work:
https://stackblitz.com/edit/github-wcewz1?file=package.json

**Reply by danielroe** [reply]
Those docs are for Nuxt 2. Check out https://v3.nuxtjs.org. `.env` variants are not supported.

**Reply by sxgrant** [reply]
Thanks Daniel.  About 15 minutes after posting that and moving on to something else I realised my mistake 🤦‍♂️
I guess since I was deep linking into that section I missed the banner at the top
